nibabel/freesurfer/io.py

```diff
@@ -400,6 +435,16 @@ def write_string(s):
             write(len(s))
             write(s, dtype='|S%d' % len(s))
 
+        # Generate annotation values for each ctab entry
+        if fill_ctab:
+            new_ctab = np.zeros((ctab.shape[0], 5), dtype=np.int32)
+            new_ctab[:, :4] = ctab[:, :4]
+            ctab = new_ctab
+            ctab[:, 4] = (ctab[:, 0] +
+                          ctab[:, 1] * (2 ** 8) +
+                          ctab[:, 2] * (2 ** 16) +
+                          ctab[:, 3] * (2 ** 24))
```

Comment:
With the refactor described above, this can just be:

```Python
if fill_ctab:
    ctab = np.hstack((ctab[:, :4], pack_rgba(ctab[:, :4])))
```

Would it also make sense to add the following?

```Python
elif not np.array_equal(ctab[:, 4], pack_rgba(ctab[:, :4])):
    # Warn of invalid array, suggest using fill_ctab
```

It seems like an error case that is worth warning about.

Reply:
Good point, will add the warning in.